src/bin/dual-iir.rs

```diff
@@ -143,23 +139,15 @@ const APP: () = {
         let clock = c.resources.clock;
 
         loop {
-            let sleep = c.resources.mqtt_interface.lock(|interface| {
+            let _sleep = c.resources.mqtt_interface.lock(|interface| {
                 !interface.network_stack().poll(clock.current_ms())
             });
 
-            match c
-                .resources
+            if c.resources
                 .mqtt_interface
                 .lock(|interface| interface.update().unwrap())
             {
-                miniconf::Action::Continue => {
-                    if sleep {
-                        cortex_m::asm::wfi();
-                    }
-                }
-                miniconf::Action::CommitSettings => {
-                    c.spawn.settings_update().unwrap()
-                }
+                c.spawn.settings_update().unwrap()
             }
```

Comment:
```suggestion
            } else {
                cortex_m::asm::wfi();
            }
```

No?

Reply:
Was just playing around with the semantics here. I think it should just be `else if sleep` - otherwise, we may sleep when other (non MQTT) network traffic is received, which I don't think would be desirable for the new streaming app?

Reply:
Yes. I hadn't seen the subsequent commits yet.